DeadCrumbs/app/src/main/java/dead/crumbs/ui/DeadReckoningService.kt

```diff
@@ -90,44 +91,34 @@ class DeadReckoningService : Service(), SensorEventListener {
     }
 
 
-    // Step detection fields
-    private var stepCounterInitial: Int = 0
-    private var stepCounter: Int = 0
+    // Step detection and step length estimation fields
+    private val accArraySize: Int = 10000
 
-    // Step length estimation fields
-    private var accelerometerZReadings = mutableListOf<Pair<Long, Float>>()
+    private var accelerometerZReadings = arrayOfNulls<Pair<Float, Double>>(accArraySize)
+    private var accBufferIndex: Int = 0
+    private var gravitationalAccel: Double = 9.72
```

Comment:
It is probably all good. I would just note that according to google the gravitational acceleration is 9.80665 m/s^2

Reply:
I updated the value :)